### PR TITLE
[ANCHOR-445] Remove unique_address callback

### PR DESCRIPTION
**Pull request by philipliu**
### Description

This removes the `unique_address` call from the SEP-31 `POST /transaction` endpoint. `stellar_account_id`, `stellar_memo`, and `stellar_memo_type` are not required fields in the SEP-31 responses, so it's not necessary to generate the deposit address at transaction creation time. This PR updates the `RequestOnchainFundsHandler` RPC to allow businesses to populate the deposit address, similar to the SEP-6 and SEP-24 flows.

### Context

We want to remove the `unique_address`  callback in 3.0.

### Testing

- `./gradlew test`

### Documentation

SEP-31 integration guide and RPC docs will be updated.

### Known limitations

N/A

### CODE REVIEW

File: core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java [hunk 1]
```diff
@@ -87,29 +79,25 @@ public Sep31Service(
...
       Sep31TransactionStore sep31TransactionStore,
-      Sep31DepositInfoGenerator sep31DepositInfoGenerator,
       Sep38QuoteStore sep38QuoteStore,
       ClientService clientService,
       AssetService assetService,
       RateIntegration rateIntegration,
-      EventService eventService,
-      CustodyService custodyService,
-      CustodyConfig custodyConfig) {
+      CustomerIntegration customerIntegration,
```
Comment (JiahuiWho) [inline core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java:86]: we can remove the `customerIntegration` if it is not being used

Comment (philipliu) [reply core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java:86]: Good catch. I missed this while rebasing.

File: core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java [hunk 2]
```diff
@@ -18,8 +18,6 @@
 import static org.stellar.anchor.util.SepHelper.validateAmount;
 import static org.stellar.anchor.util.SepHelper.validateAmountLimit;
 import static org.stellar.anchor.util.SepLanguageHelper.validateLanguage;
-import static org.stellar.anchor.util.StringHelper.isEmpty;
-import static org.stellar.sdk.xdr.MemoType.MEMO_NONE;
 
 import io.micrometer.core.instrument.Counter;
 import jakarta.transaction.Transactional;
```
Comment (lijamie98) [inline core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java:23]: Jakarta is old. Should we use `javax.transaction.Transactional`?

File: core/src/test/kotlin/org/stellar/anchor/sep31/Sep31ServiceTest.kt [hunk 3]
```diff
@@ -743,7 +737,7 @@ class Sep31ServiceTest {
     val wantTx =
```
Comment (lijamie98) [inline core/src/test/kotlin/org/stellar/anchor/sep31/Sep31ServiceTest.kt:737]: `memo` in L735 is assigned twice.

**Review by lijamie98 — CHANGES_REQUESTED** [review]

File: core/src/test/kotlin/org/stellar/anchor/sep31/Sep31ServiceTest.kt (hunk 3 above)
Comment (lijamie98) [reply core/src/test/kotlin/org/stellar/anchor/sep31/Sep31ServiceTest.kt:737]: In https://github.com/stellar/java-stellar-anchor-sdk/pull/1446/files#diff-4e0835ca8055b8e40053d5f39ecebc0b44ac62e037db1367d0d3d639949a4dd3L4

Can we use a different package?

File: platform/src/main/java/org/stellar/anchor/platform/component/platform/PlatformServerBeans.java [hunk 4]
```diff
@@ -66,6 +69,27 @@ public FilterRegistrationBean<Filter> platformTokenFilter(PlatformServerConfig c
...
+        return new Sep31DepositInfoSelfGenerator();
+      case CUSTODY:
+        return new Sep31DepositInfoCustodyGenerator(
+            custodyApiClient.orElseThrow(
+                () ->
+                    new InvalidConfigException("Integration with custody service is not enabled")));
+      case NONE:
+        return new Sep31DepositInfoNoneGenerator();
+      default:
+        throw new RuntimeException("Not supported");
```
Comment (lijamie98) [inline platform/src/main/java/org/stellar/anchor/platform/component/platform/PlatformServerBeans.java:87]:
```suggestion
        throw new RuntimeException("sep31.deposit_info_generator_type: ${sep31Config.getDepositInfoGeneratorType()} is not supported");
```

File: core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java (hunk 2 above)
Comment (philipliu) [reply core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java:23]: `javax` was renamed to `jakarta` https://mvnrepository.com/artifact/javax.transaction/jta

File: core/src/test/kotlin/org/stellar/anchor/sep31/Sep31ServiceTest.kt (hunk 3 above)
Comment (philipliu) [reply core/src/test/kotlin/org/stellar/anchor/sep31/Sep31ServiceTest.kt:737]: > memo in L735 is assigned twice.

Nice catch, this variable is no longer used anymore. I deleted it.

Comment (philipliu) [reply core/src/test/kotlin/org/stellar/anchor/sep31/Sep31ServiceTest.kt:737]: > In https://github.com/stellar/java-stellar-anchor-sdk/pull/1446/files#diff-4e0835ca8055b8e40053d5f39ecebc0b44ac62e037db1367d0d3d639949a4dd3L4
>
> Can we use a different package?

The link doesn't seem to work. Can you clarify what you mean?

Comment (lijamie98) [reply core/src/test/kotlin/org/stellar/anchor/sep31/Sep31ServiceTest.kt:732]: ah... I cannot recall. It is minor. May be will catch next time.

**Review by lijamie98 — APPROVED** [review]
LGTM

**Pull request merged**